taichi/ir/snode.cpp

```diff
@@ -312,6 +351,11 @@ SNode *SNode::get_adjoint() const {
   return grad_info->adjoint_snode();
 }
 
+SNode *SNode::get_adjoint_visited() const {
+  // TI_ASSERT(has_adjoint());
```

Comment:
OOC: is this needed?

Reply:
It will be used when loading or storing to the adjoint_visited SNode.